packages/next/src/build/webpack/plugins/flight-client-entry-plugin.ts

```diff
@@ -296,10 +300,14 @@ export class FlightClientEntryPlugin {
         compilation.moduleGraph
       )) {
         // Entry can be any user defined entry files such as layout, page, error, loading, etc.
-        const entryRequest = (
+        let entryRequest = (
           connection.dependency as unknown as webpack.NormalModule
         ).request
 
+        if (entryRequest.endsWith(WEBPACK_RESOURCE_QUERIES.metadataRoute)) {
```

Comment:
Not blocking: would be nice to have a helper to extract entryRequest and handle all cases, using `let` scares me that we might need to take care of these cases everywhere when we create them.